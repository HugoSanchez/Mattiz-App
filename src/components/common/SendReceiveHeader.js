import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

import styles from '../../styles';
import colors from '../../constants/colors';

// Props:
// @ onSendPress:    - Required. Function to be executed.
// @ onReceivePress: - Required. Function to be executed.
// @ isSendSelected: - Required. Boolean.

const SendReceiveHeader = props => {

    const {
        extraSmallMediumTitle,
        extraSmallLightTitle
    } = styles;

    const { 
        container, 
        selectedStyle,
    } = compStyles;

    const {
        isSendSelected
    } = props;

    return (
        <View style={ container }>
            <View style={ styles.container }>
                <TouchableOpacity
                    style={ isSendSelected ? selectedStyle : null} 
                    onPress={ props.onSendPress }> 
                    <Text style={ isSendSelected ? extraSmallMediumTitle : extraSmallLightTitle }> 
                        Send 
                    </Text>
                </TouchableOpacity>
            </View>
            <View style={ styles.container }>
                <TouchableOpacity 
                    style={ !isSendSelected ? selectedStyle : null}
                    onPress={ props.onReceivePress }>
                    <Text style={ !isSendSelected ? extraSmallMediumTitle : extraSmallLightTitle }>
                        Receive 
                    </Text>
                </TouchableOpacity>
            </View>
        </View>
    );
}

const compStyles = {
    container: { 
        flex: 1, 
        flexDirection: 'row' 
    },
    selectedStyle: { 
        borderBottomWidth: 1, 
        borderColor: colors.primaryBlue 
    }
}

export default SendReceiveHeader;