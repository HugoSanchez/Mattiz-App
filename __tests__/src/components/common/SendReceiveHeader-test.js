import 'react-native'
import React from 'react'
import SendReceiveHeader from '../../../../src/components/common/SendReceiveHeader'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<SendReceiveHeader />)
	expect(container.children[0]).toMatchSnapshot()
})
