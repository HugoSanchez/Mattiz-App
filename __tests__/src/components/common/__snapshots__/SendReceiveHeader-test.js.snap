// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 2,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": "4%",
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 25,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïˆ‡
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={null}
      >
        <Text
          style={
            Object {
              "color": "#154053",
              "fontFamily": "Rajdhani-Light",
              "fontSize": 18,
            }
          }
        >
          Send
        </Text>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": undefined,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#154053",
              "fontFamily": "Raleway-Medium",
              "fontSize": 18,
            }
          }
        >
          Receive
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
